install/installer/pkg/config/v1/experimental/experimental.go

```diff
@@ -140,6 +140,7 @@ type ServerConfig struct {
 	GithubApp                         *GithubApp          `json:"githubApp"`
 	ChargebeeSecret                   string              `json:"chargebeeSecret"`
 	StripeSecret                      string              `json:"stripeSecret"`
+	StripeConfig                      string              `json:"stripeConfig"`
```

Comment:
I'm not sure if this is just me failing to understand, but should this value be a string? From what I can tell by reading the code, this gets injected into the configmap and then sent to the `server`.

According to `components/server/src/config.ts`, the definition is:

```
stripeConfig?: { usageProductPriceIds: { EUR: string; USD: string } };
```

I can see you're using the file, but I can't see how the value is anything other than an object that you're passing in as a string?

To my thinking, it would be better to define that as a Golang struct and pass it in as YAML and let the Installer parse it to a Golang struct and then inject it that way.

If I've mistaken what's going on, apologies. If you want a short sync so we can talk it over, let me know

Reply:
I'm not a 100% sure how this works either, but isn't this a raw JSON string, which we then attempt to parse in the server config?

Reply:
Maybe `StripeConfigMap`? But does not need to change.

Reply:
> To my thinking, it would be better to define that as a Golang struct and pass it in as YAML and let the Installer parse it to a Golang struct and then inject it that way.

Yes, that would be one way, but also tie the configs together. But we'd rather have them independent for several reasons. Happy to align afterwards, but I'd be super happy to see this merged! :orange_heart:

Reply:
As `experimental` config is owned by the team it appears under, I'm happy to approve it. I can see the benefit of not tying the config together. I'll approve it on behalf to Self Hosted.